rmgpy/data/thermo.py

```diff
@@ -1414,16 +1441,42 @@ def computeGroupAdditivityThermo(self, molecule):
                     logging.error(molecule.toAdjacencyList())
                     raise
                 # Correct for gauche and 1,5- interactions
-                if not cyclic:
-                    try:
-                        self.__addGroupThermoData(thermoData, self.groups['gauche'], molecule, {'*':atom})
-                    except KeyError: pass
-                try:
-                    self.__addGroupThermoData(thermoData, self.groups['int15'], molecule, {'*':atom})
-                except KeyError: pass
+                # Pair atom with its 1st and 2nd nonHydrogen neighbors, 
+                # Then match the pair with the entries in the database longDistanceInteraction_noncyclic.py
+                # Currently we only have gauche(1,4) and 1,5 interactions in that file. 
+                # If you want to add more corrections for longer distance, please call getNthNeighbor() method accordingly.
+                # Potentially we could include other.py in this database, but it's a little confusing how to label atoms for the entries in other.py
+                if not molecule.isAtomInCycle(atom):
+                    for atom_2 in molecule.getNthNeighbor([atom],1,[])+molecule.getNthNeighbor([atom],2,[]):
```

Comment:
I think there's potential here to have `getNthNeighbor` accept a list of distances and return all neighbors at those distances. For example, since getting distance 2 neighbors requires finding distance 1 neighbors, the current way actually requires 3 function calls, one of which is repeated. I'll comment on the actual function with a code suggestion.

While it's not that big of a deal for distance 1 and 2 neighbors, it would help for future extensibility if people add corrections for longer distances.